libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -402,9 +402,8 @@ bool AP_DDS_Client::start(void)
 }
 
 // read function triggered at every subscription callback
-void AP_DDS_Client::on_topic (uxrSession* uxr_session, uxrObjectId object_id, uint16_t request_id, uxrStreamId stream_id, struct ucdrBuffer* ub, uint16_t length, void* args)
+void AP_DDS_Client::on_topic ([[maybe_unused]] uxrSession* uxr_session, uxrObjectId object_id, [[maybe_unused]] uint16_t request_id, [[maybe_unused]] uxrStreamId stream_id, struct ucdrBuffer* ub, [[maybe_unused]] uint16_t length, void* args)
```

Comment:
Should we be using an `MAYBE_UNUSED_PARAMETER` macro instead here, matching all of our other `*UNUSED*` macros from AP_Common.h?

Reply:
https://github.com/ArduPilot/ardupilot/pull/17779#discussion_r653224220